### PR TITLE
add a function to check the existence of mesh meta data and add mesh meta data

### PR DESCRIPTION
**Pull request by YaqiWang**
Refer to #15987.

I am not closing #15987 because we need that issue for removing unnecessary functions in `MooseMesh` once we removed all mesh modifiers.

### CODE REVIEW

File: framework/src/base/MooseApp.C [hunk 1]
```diff
@@ -1206,6 +1206,28 @@ MooseApp::registerRestartableData(const std::string & name,
   return *insert_pair.first->second.value;
 }
 
+bool
+MooseApp::hasRestartableData(const std::string & name,
```
Comment (aeslaughter) [inline framework/src/base/MooseApp.C:1210]: Rather than have these error checks, can you just create a hasRestartableMetaData method?

**Review by aeslaughter — CHANGES_REQUESTED** [review]
Please add test(s) that use these new functions.

File: framework/src/base/MooseApp.C (hunk 1 above)
Comment (YaqiWang) [reply framework/src/base/MooseApp.C:1210]: That sounds cleaner. The only issue is that this function will be different from `registerRestartableData`, which put meta data and normal data together. I slightly prefer two separate functions. If you still want this change. I can certainly do.

**Comment by YaqiWang** [comment]
@aeslaughter Comments addressed. Should be ready.

**Review by aeslaughter — DISMISSED** [review]

**Comment by YaqiWang** [comment]
Sorry, noticed a few mistakes in doxygen comments.

**Review by aeslaughter — APPROVED** [review]

**Review by permcody — APPROVED** [review]

**Comment by YaqiWang** [comment]
Thanks!

**Pull request merged**